comment/api/serializers.py

```diff
@@ -99,25 +99,21 @@ class Meta:
         model = Comment
         fields = ('id', 'user', 'email', 'content', 'parent', 'posted', 'edited', 'reply_count', 'replies', 'urlhash')
 
-    def __init__(self, *args, **kwargs):
-        user = kwargs['context']['request'].user
-        self.email_service = None
-        if user.is_authenticated or not settings.COMMENT_ALLOW_ANONYMOUS:
-            del self.fields['email']
-
-        super().__init__(*args, **kwargs)
-
     @staticmethod
-    def validate_email(value):
-        if not value:
-            raise serializers.ValidationError(EmailError.EMAIL_MISSING, code='required')
-        return value.strip().lower()
+    def validate_email(email):
+        if not email:
+            raise serializers.ValidationError(
+                detail={'email': [EmailError.EMAIL_REQUIRED_FOR_ANONYMOUS]}, code='required'
+            )
+        return email.strip().lower()
 
     def create(self, validated_data):
         request = self.context['request']
         user = get_user_for_request(request)
         content = validated_data.get('content')
         email = validated_data.get('email')
```

Comment:
`validate_email` is already called here. You can try to enter a email like `A@A.COM` with an anonymous user and see the corresponding value in the database. If the value saved inside the database is `a@a.com`, you won't need to call `validate_email` manually.